app/backend/chat_history/cosmosdb.py

```diff
@@ -0,0 +1,198 @@
+import os
+import time
+from typing import Any, Dict, Union
+
+from azure.cosmos.aio import ContainerProxy, CosmosClient
+from azure.identity.aio import AzureDeveloperCliCredential, ManagedIdentityCredential
+from quart import Blueprint, current_app, jsonify, request
+
+from config import (
+    CONFIG_CHAT_HISTORY_COSMOS_ENABLED,
+    CONFIG_COSMOS_HISTORY_CLIENT,
+    CONFIG_COSMOS_HISTORY_CONTAINER,
+    CONFIG_CREDENTIAL,
+)
+from decorators import authenticated
+from error import error_response
+
+chat_history_cosmosdb_bp = Blueprint("chat_history_cosmos", __name__, static_folder="static")
+
+
+@chat_history_cosmosdb_bp.post("/chat_history")
+@authenticated
+async def post_chat_history(auth_claims: Dict[str, Any]):
+    if not current_app.config[CONFIG_CHAT_HISTORY_COSMOS_ENABLED]:
+        return jsonify({"error": "Chat history not enabled"}), 405
+
+    container: ContainerProxy = current_app.config[CONFIG_COSMOS_HISTORY_CONTAINER]
```

Comment:
This is called in every function and I assume it never changes. Could this be cached or defined globally?

Reply:
`current_app.config` is basically our global dict, it's how we access objects that were setup at the beginning of the app start. I don't think there's a performance hit, since it should be O(1) retrieval.